Base/usr/share/man/man5/ipc.md

```diff
@@ -1,17 +1,84 @@
 ## Name
 
-ipc - Inter Process Communication endpoint definition format
+IPC - Inter-Process Communication endpoint definition format (.ipc)
 
 ## Synopsis
 
 The IPC format of SerenityOS is a domain-specific language (DSL) used to define communication endpoints for IPC.
 
 ## Description
 
-These files are human-readable, have no easily detectable filemagic, and define IPC interfaces.
-The format is loosely inspired by C++ headers.
+Informally, IPC files - with the help of the IPC compiler - are used to generate message classes that will **wrap**
+messages for interprocess communication in the system. IPC syntax is loosely inspired by C++ headers. Generated IPC
+message classes support encode and decode functions to pass messages between the processes.
 
-## See Also
+Every IPC pair in the system has a client endpoint and a server endpoint that are described in the IPC files.
+Each IPC endpoint should have a unique hashable name that will uniquely identify endpoints in the system.
 
-- [`Meta/Lagom/Tools/CodeGenerators/IPCCompiler/`](../../../../../Meta/Lagom/Tools/CodeGenerators/IPCCompiler/).
-- [ipc(4)](help://man/4/ipc) (IPC Unix socket documentation)
+There are 2 types of APIs that are supported by the IPC files: synchronous and asynchronous.
+Synchronous function calls always wait for a response from the other side, while the asynchronous counterparts do not.
+In other words, in case of the synchronous calls, the IPC library will not return until it has a response for a caller.
+
+Ideally, all APIs for the server endpoint should be asynchronous.
+
+## Examples
+
+To create a new connection, you first need to generate client and server endpoints.
+These endpoints should implement the communication logic using the IPC compiler-generated API messages.
+
+Start from defining an endpoint in the IPC file in MyServer.ipc.
+
+```
+endpoint MyServer
+{
+    SyncAPI(String text) => (i32 status)
+    AsyncAPI(i32 mode) =|
+}
+```
+
+Part of the generated C++ messages
+
+```c++
+class SyncAPI final : public IPC::Message {
+public:
+    using ResponseType = SyncAPIResponse;
+    SyncAPI(const String& text) : m_text(text) {}
+    virtual ~SyncAPI() override {}
+    static OwnPtr<SyncAPI> decode(/**/);
+    virtual IPC::MessageBuffer encode(/**/) const override;
```

Comment:
I probably wasn't clear enough with my previous comment; using `...` instead of `/**/` for omission is completely OK but the missing semicolons were the thing that threw me off.

Reply:
It was clear. Now everything seems to be syntactically valid, or not?

Reply:
No, you should use `...` for omission as is done in other code snippets.